### PR TITLE
Logging level, token bug fixes, and error handling

### PR DESCRIPTION
**Pull request by ershockley**
This PR includes

1. Adds a `logging_level` attribute to the utilix Config class, so that you can set the logging level in your config. I already updated the global config level on midway. It defaults to 'WARNING' if no option is passed. 

2. Cleaned up the Token class more, hopefully resolving all the bugs we're seeing with authentication. It's a bit difficult to test every scenario, but it resolves @SidAhmedMa's issue. 

3. Generally better error handling. Again I'm not sure I've satisfied everything, but hopefully the error messages will be more helpful now. For example, bad API calls raise a RuntimeError so that we don't get that useless `KeyError: 'results' not in the dict` error downstream.

### CODE REVIEW

File: utilix/rundb.py [hunk 1]
```diff
@@ -116,6 +129,10 @@ class Token:
     Object handling tokens for runDB API access.
     
     """
+    string = None
```
Comment (jorana) [inline utilix/rundb.py:132]: This ` string` is the `token_string` or something that the rundb api gives you to access the rundb api? Perhaps a comment (and/or somewhat more helpful variable name) will help others understand what is going on.

**Review by jorana — APPROVED** [review]
Hi Evan, thanks for changing this. I'm not sure if I understand enough of the rundb api to fully grasp where things improved.

For sure, the improved error handling and grabbing the `results` will help!

File: utilix/rundb.py [hunk 2]
```diff
@@ -153,46 +172,52 @@ def new_token(self):
...
         response = requests.get(url, headers=headers)
-        response = json.loads(response.text)
-
-        # if rewew fails, try logging back in
-        if (response['status_code'] is not 200 and response['error'] != 'EarlyRefreshError'):
-            print("Refreshing token")
-            self.string, self.user = self.new_token()
+        logger.debug(f"The response was {response.text}")
+        # if renew fails, try logging back in
+        if response.status_code is not 200:
```
Comment (jorana) [inline utilix/rundb.py:209]: Maybe also up this syntax:
afbeelding

**Comment by ershockley** [comment]
Thanks for the comments @jorana. I'm adressing them now. Just to give you a better idea of the improvement:

Previously:
```
>>> import utilix
>>> utilix.db.get_context('shit', '0.12.3')
2020-12-04 11:43:13,886 - utilix - ERROR - HTTP(s) request says: not_found (Code 404)
{}
```


Now: 

```
>>> import utilix
>>> utilix.db.get_context('shit', '0.12.3')
2020-12-04 11:42:33,307 - utilix - ERROR - API Call was /contexts/0_12_3/shit/: HTTP(s) request says: not_found (Code 404)
2020-12-04 11:42:33,308 - utilix - ERROR - Error 404 means the API call was not formatted correctly. Check the URL.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ershockley/software/utilix/utilix/rundb.py", line 406, in get_context
    response = json.loads(self._get(url).text)
  File "/home/ershockley/software/utilix/utilix/rundb.py", line 120, in func_wrapper
    raise RuntimeError("API call failed.")
RuntimeError: API call failed.
```

File: utilix/rundb.py [hunk 3]
```diff
@@ -10,8 +10,10 @@
 
 #Config the logger:
 logger = logging.getLogger("utilix")
+level = logging.DEBUG
```
[the remark this answers is not in the document]
Comment (ershockley) [reply utilix/rundb.py]: Thanks for the catch. Removed/

File: utilix/rundb.py (hunk 1 above)
Comment (ershockley) [reply utilix/rundb.py]: Yes it is the token string (I thought of it as Token.string). Modified the name to token_string

File: utilix/rundb.py (hunk 2 above)
Comment (ershockley) [reply utilix/rundb.py]: done

**Comment by ershockley** [comment]
@jorana I know you approved already but can you give me a quick thumbs up if I can merge?

**Comment by jorana** [comment]
Thanks looks good Evan!

**Pull request merged**